src/Raven.Client/Documents/Operations/AI/VertexSettings.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.AI;
+
+public sealed class VertexSettings : AbstractAiSettings
+{
+    public VertexSettings()
+    {
+        // deserialization
+    }
+    
+    public VertexSettings(string model, string googleCredentialsJson, string location, string projectId, VertexAIVersion? aiVersion = null)
+    {
+        Model = model;
+        GoogleCredentialsJson = googleCredentialsJson;
+        Location = location;
+        ProjectId = projectId;
+        AiVersion = aiVersion;
+    }
+    
+    /// <summary>
+    /// The model ID for the Vertex AI service.
+    /// </summary>
+    public string Model { get; set; }
+    
+    public string GoogleCredentialsJson { get; set; }
+    
+    public VertexAIVersion? AiVersion { get; set; }
+    
+    public string Location { get; set; }
+    
+    public string ProjectId { get; set; }
```

Comment:
check if this shouldn't be extracted from the GoogleCredentialsJson? We are already doing that in the RavenGoogleCloudClient